lib/api/v2/WebSocketHandler.ts

```diff
@@ -0,0 +1,190 @@
+import http from 'http';
+import ws from 'ws';
+import { formatError } from '../../Utils';
+import Service from '../../service/Service';
+import Controller from '../Controller';
+import { apiPrefix } from './Consts';
+
+enum Operation {
+  Ping = 'ping',
+  Pong = 'pong',
+
+  // TODO: unsubscribe
+  Subscribe = 'subscribe',
+  Update = 'update',
+}
+
+enum SubscriptionChannel {
+  SwapUpdate = 'swap.update',
+}
+
+type WsRequest = {
+  op: Operation;
+};
+
+type WsSubscribeRequest = WsRequest & {
+  channel: SubscriptionChannel;
+  args: string[];
+};
+
+type WsResponse = {
+  event: Operation;
+};
+
+type WsErrorResponse = {
+  error: string;
+};
+
+class WebSocketHandler {
+  private readonly ws: ws.Server;
+
+  private readonly swapToSockets = new Map<string, ws[]>();
```

Comment:
```suggestion
  private readonly swapToSocket = new Map<string, ws>();
```
Allow just one socket per swap?